app/javascript/packages/document-capture/context/acuant.tsx

```diff
@@ -0,0 +1,270 @@
+  sdkSrc: string; // SDK source URL.
+  cameraSrc: string; // Camera JavaScript source URL.
+  credentials: string | null; // SDK credentials.
+  endpoint: string; // Endpoint to submit payload.
+  glareThreshold: number; // Minimum acceptable glare score for images.
+  sharpnessThreshold: number; // Minimum acceptable sharpness score for images.
+  children: ReactNode; // Child element
+}
+
+/**
+ * The minimum glare score value to be considered acceptable.
+ *
+ */
+export const DEFAULT_ACCEPTABLE_GLARE_SCORE = 30;
+
+/**
+ * The minimum sharpness score value to be considered acceptable.
+ *
+ */
+export const DEFAULT_ACCEPTABLE_SHARPNESS_SCORE = 30;
+
+/**
+ * Returns the containing directory of the given file, including a trailing slash.
+ */
+export const dirname = (file: string): string => {
+  return file.split('/').slice(0, -1).concat('').join('/');
+};
+
+interface AcuantContextInterface {
+  isReady: boolean;
+  isAcuantLoaded: boolean;
+  isError: boolean;
+  isCameraSupported: boolean | null;
+  isActive: boolean;
+  setIsActive: (nextIsActive: boolean) => void;
+  credentials: string | null;
+  glareThreshold: number;
+  sharpnessThreshold: number;
+  endpoint: string | null;
+}
+
+const AcuantContext = createContext<AcuantContextInterface>({
+  isReady: false,
+  isAcuantLoaded: false,
+  isError: false,
+  isCameraSupported: /** @type {boolean?} */ null,
+  isActive: false,
+  setIsActive: () => {},
+  credentials: null,
+  glareThreshold: DEFAULT_ACCEPTABLE_GLARE_SCORE,
+  sharpnessThreshold: DEFAULT_ACCEPTABLE_SHARPNESS_SCORE,
+  endpoint: /** @type {string?} */ null,
+});
+
+AcuantContext.displayName = 'AcuantContext';
+
+/**
+ * Returns a found AcuantJavascriptWebSdk
+ * object, if one is available.
+ * This function normalizes differences between
+ * the 11.5.0 and 11.7.0 SDKs. The former attached
+ * the object to the global window, while the latter
+ * sets the object in the global (but non-window)
+ * scope.
+ */
+const getActualAcuantJavascriptWebSdk = (): AcuantJavascriptWebSdkInterface => {
+  if (window.AcuantJavascriptWebSdk) {
+    return window.AcuantJavascriptWebSdk;
+  }
+  return AcuantJavascriptWebSdk;
+};
+
+/**
+ * Returns a found AcuantCamera
+ * object, if one is available.
+ * This function normalizes differences between
+ * the 11.5.0 and 11.7.0 SDKs. The former attached
+ * the object to the global window, while the latter
+ * sets the object in the global (but non-window)
+ * scope.
+ */
+const getActualAcuantCamera = (): AcuantCameraInterface => {
+  if (window.AcuantCamera) {
+    return window.AcuantCamera;
+  }
+  return AcuantCamera;
+};
+
+function AcuantContextProvider<AcuantContextProviderProps>({
+  sdkSrc = '/acuant/11.7.0/AcuantJavascriptWebSdk.min.js',
+  cameraSrc = '/acuant/11.7.0/AcuantCamera.min.js',
+  credentials = null,
+  endpoint = null,
+  glareThreshold = DEFAULT_ACCEPTABLE_GLARE_SCORE,
+  sharpnessThreshold = DEFAULT_ACCEPTABLE_SHARPNESS_SCORE,
+  children,
+}) {
```

Comment:
This generic isn't working the way you might be expecting in assigning types of the arguments.

![image](https://user-images.githubusercontent.com/1779930/181595705-cb7d49bf-5be8-406f-9ab6-948a9a10bdb1.png)

```suggestion
function AcuantContextProvider({
  sdkSrc = '/acuant/11.7.0/AcuantJavascriptWebSdk.min.js',
  cameraSrc = '/acuant/11.7.0/AcuantCamera.min.js',
  credentials = null,
  endpoint = null,
  glareThreshold = DEFAULT_ACCEPTABLE_GLARE_SCORE,
  sharpnessThreshold = DEFAULT_ACCEPTABLE_SHARPNESS_SCORE,
  children,
}: AcuantContextProviderProps) {
```

Reply:
Updated in b563c26